frontend/app/src/features/matching/components/SelectPreferencesCard.tsx

```diff
@@ -44,6 +79,39 @@ export const SelectPreferencesCard = ({ joinCallback }: Props) => {
       topics: [],
     },
   });
+  const watchDifficulties = watch("difficulties");
+  const watchTopics = watch("topics");
+  const prevTopicsLength = useRef(watchTopics.length);
+  const { isLoading, data } = useGetQuestionFilters();
+
+  const difficultyOptions = useMemo(() => {
+    return Object.values(DIFFICULTY).map(difficulty => ({
+      value: difficulty,
+      label: difficulty,
+    }));
+  }, []);
+
+  const topicOptions = useMemo(
+    () => getTopicsForDifficulty(watchDifficulties, data?.data),
+    [watchDifficulties],
+  );
+
+  // Filter user selected topics when difficulty changes
+  useEffect(() => {
```

Comment:
nit: not v good to use a `useEffect` like that. the 'effect' (to update the value) should be set in the change handler instead.

but tbh if it works idrc LOL up to u

Reply:
got rid of useeffect just using a const now